cpp/oneapi/dal/algo/decision_forest/backend/gpu/infer_auxiliary_structs.hpp

```diff
@@ -0,0 +1,73 @@
+/*******************************************************************************
+* Copyright 2021 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#pragma once
+
+#include "oneapi/dal/algo/decision_forest/common.hpp"
+#include "oneapi/dal/detail/common.hpp"
+#include "oneapi/dal/backend/primitives/ndarray.hpp"
+
+namespace oneapi::dal::decision_forest::backend {
+
+template <typename Index, typename Task>
+struct infer_impl_const;
+
+template <typename Index>
+struct infer_impl_const<Index, task::classification> {
+    constexpr static Index bad_val_ = -1;
+    constexpr static Index leaf_mark_ = bad_val_;
+};
+
+template <typename Index>
+struct infer_impl_const<Index, task::regression> {
+    constexpr static Index bad_val_ = -1;
+    constexpr static Index leaf_mark_ = bad_val_;
+};
```

Comment:
`bad_val_` is the same for classification and regression, is it expected?

Reply:
yes

Reply:
Specialization is done just in case of changing those values in the future?

Reply:
correct